### ISSUE TITLE
Update sapper docs to conform with Helmet v4

### ISSUE THREAD

**Issue by mhatvan** [sveltejs/sapper#1361]
A new major version of [Helmet](https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md) was released a couple days ago and the [Sapper documentation about Content Security Policy](https://sapper.svelte.dev/docs#Content_Security_Policy_CSP) needs to be adapted to it. One thing that changed for example is that "you can no longer use a function inside a directive, e.g. for nonce generation". 

I was not able to set up a working nonce generation for my project to avoid CSP issues, there is also a recent issue that might be relevant: https://github.com/helmetjs/helmet/issues/237

**Comment by mhatvan** [comment]
Yes, Helmet 4.1.0 re-introducted functions for nonce generation, but there are more issues since it is way more strict.

I just tried to make it work again, no chance.

The docs need to adapted to at least set `default-src`, otherwise running into `Error: Content-Security-Policy needs a default-src but none was provided`. You can see here in the changelog: https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md#added-1

Other than that, I don't see the nonce being set in the DOM, just `<script nonce>` without any value.

And this is my console output after having `script-src` and `default-src` set:

![2020-09-04_22-10](https://user-images.githubusercontent.com/16797721/92281325-9c9da300-eefb-11ea-832d-1f476a0097a4.png)

Some errors I can guess how to fix, others not.

**Comment by benmccann** [comment]
I think "Refused to apply the inline style" message would be fixed by https://github.com/sveltejs/sapper/pull/1232. I just did another pass at reviewing that one, so hopefully we can merge it soon

The others look like they're mostly not things that Sapper controls. You might need https://github.com/sveltejs/sapper/pull/1249 for those.

It seems like you'd run into these problems regardless of Helmet version. Are you okay if we leave this closed since there are already other issues tracking those problems?

**Comment by mhatvan** [comment]
We can keep it closed, no problem!
Thank you for pushing these CSP issues and pull requests forward, would be great to get the implementation finished up.

**Comment by mhatvan** [comment]
I was able to get CSP working after all the related issues were fixed and merged.

I just wanted to leave my helmet configuration here, in case anyone doesn't have the time to figure this all out alone:
```js
app
  .use(
    helmet({
      contentSecurityPolicy: {
        directives: {
          defaultSrc: ["'self'"],
          // Has to be unsafe-eval because %sapper.scripts% uses eval
          // @ts-expect-error
          scriptSrc: ["'self' 'unsafe-eval'", (_req, res) => `'nonce-${res.locals.nonce}'`],
          // @ts-expect-error
          scriptSrcElem: ["'self' blob:", (_req, res) => `'nonce-${res.locals.nonce}'`],
          // Has to be unsafe-inline currently, because svelte-awesome & svelte-image sets inline style
          styleSrc: ["'self' 'unsafe-inline'"],
          frameAncestors: 'frame.bloglovin.com',
          imgSrc: ["'self'", 'data:'],
          connectSrc: ["'self'", 'http://localhost:10000'],
        },
      },
    }),
    compression({ threshold: 0 }),
    sirv('static', { dev }),
    sapper.middleware(),
  )
```

**Comment by 7antra** [comment]
> I was able to get CSP working after all the related issues were fixed and merged.
> 
> I just wanted to leave my helmet configuration here, in case anyone doesn't have the time to figure this all out alone:
> 
> ```js
> app
>   .use(
>     helmet({
>       contentSecurityPolicy: {
>         directives: {
>           defaultSrc: ["'self'"],
>           // Has to be unsafe-eval because %sapper.scripts% uses eval
>           // @ts-expect-error
>           scriptSrc: ["'self' 'unsafe-eval'", (_req, res) => `'nonce-${res.locals.nonce}'`],
>           // Has to be unsafe-inline currently, because svelte-awesome & svelte-image sets inline style
>           styleSrc: ["'self' 'unsafe-inline'"],
>           // data: needed for svelte-image placeholders and svelte-awesome icons
>           imgSrc: ["'self'", 'data:'],
>           // localhost:10000 needed by __sapper__ itself
>           connectSrc: ["'self'", 'http://localhost:10000'],
>         },
>       },
>     }),
>     compression({ threshold: 0 }),
>     sirv('static', { dev }),
>     sapper.middleware(),
>   )
> ```
Hello @mhatvan 
Did you test it on safari ?

**Comment by ejoflo** [comment]
> I was able to get CSP working after all the related issues were fixed and merged.
> 
> I just wanted to leave my helmet configuration here, in case anyone doesn't have the time to figure this all out alone:
> 
> ```js
> app
>   .use(
>     helmet({
>       contentSecurityPolicy: {
>         directives: {
>           defaultSrc: ["'self'"],
>           // Has to be unsafe-eval because %sapper.scripts% uses eval
>           // @ts-expect-error
>           scriptSrc: ["'self' 'unsafe-eval'", (_req, res) => `'nonce-${res.locals.nonce}'`],
>           // Has to be unsafe-inline currently, because svelte-awesome & svelte-image sets inline style
>           styleSrc: ["'self' 'unsafe-inline'"],
>           // data: needed for svelte-image placeholders and svelte-awesome icons
>           imgSrc: ["'self'", 'data:'],
>           // localhost:10000 needed by __sapper__ itself
>           connectSrc: ["'self'", 'http://localhost:10000'],
>         },
>       },
>     }),
>     compression({ threshold: 0 }),
>     sirv('static', { dev }),
>     sapper.middleware(),
>   )
> ```

Thanks for this, it really saved me a lot of headaches!
